tests/bsim/bluetooth/mesh/src/test_suspend.c

```diff
@@ -464,6 +464,34 @@ static void test_tester_gatt(void)
 	PASS();
 }
 
+static void test_dut_suspend_resume_unref(void)
+{
+	const struct bt_mesh_test_cfg tx_cfg = {
+		.addr = 0x0001,
+		.dev_key = {0x01},
+	};
+
+	bt_mesh_test_cfg_set(&tx_cfg, 2 * WAIT_TIME);
+	bt_mesh_test_setup();
+
+	for (uint16_t count = 0; count < 2 * CONFIG_BT_MESH_ADV_BUF_COUNT; count++) {
+		LOG_DBG("Step %d", count);
+
+		ASSERT_OK_MSG(bt_mesh_test_send_data(BT_MESH_ADDR_ALL_NODES, NULL,
+						     (uint8_t *)&count, sizeof(count), NULL, NULL),
+			      "Cannot send data");
+		ASSERT_OK_MSG(bt_mesh_test_send_data(BT_MESH_ADDR_ALL_NODES, NULL,
+						     (uint8_t *)&count, sizeof(count), NULL, NULL),
+			      "Cannot send data");
+
+		ASSERT_OK_MSG(bt_mesh_suspend(), "Failed to suspend Mesh.");
+		k_sleep(K_MSEC(500));
```

Comment:
I'm wondering if this test will actually catch the issue. Since sleep is called for 500ms, the `bt_mesh_test_send_data()` will enqueue the PDU, and then PDU will be disposed off by the advertiser. So, this will always pass.

What would be interesting to check, is:
1. what happens when you have `k_sleep(K_MSEC(5))`  OR
2. have model retransmissions set to max (7), and network retransmissions set to max (7), 

Will this still pass in such scenarios?

Reply:
I did tdd. Test was implemented first and it was failed. After my initial implementation, it passed. After Pavel's clarification it will not clear adv for the sequenced sending, I added it and test was failed again. After applying Pavel's suggestions, it started passing again.
Answer: yes - test actually catches the issue.